src/Akka.Cluster.Hosting/SBR/SplitBrainResolverOption.cs

```diff
@@ -0,0 +1,99 @@
+    /// Note that you must not add more members to the cluster than '<see cref="QuorumSize"/> * 2 - 1', because then
+    /// both sides may down each other and thereby form two separate clusters. For example, <see cref="QuorumSize"/>
+    /// configured to 3 in a 6 node cluster may result in a split where each side consists of 3 nodes each,
+    /// i.e. each side thinks it has enough nodes to continue by itself. A warning is logged if this recommendation is violated.
+    /// </para>
+    /// </summary>
+    public sealed class StaticQuorumOption : SplitBrainResolverOption
+    {
+        /// <summary>
+        /// Minimum number of nodes that the cluster must have
+        /// </summary>
+        public int QuorumSize { get; set; } = 0;
+    }
+
+    /// <summary>
+    /// Down the unreachable nodes if the current node is in the majority part based the last known membership
+    /// information. Otherwise down the reachable nodes, i.e. the own part. If the the parts are of equal size the part
+    /// containing the node with the lowest address is kept.
+    /// Note that if there are more than two partitions and none is in majority each part will shutdown itself,
+    /// terminating the whole cluster.
+    /// </summary>
+    public sealed class KeepMajorityOption : SplitBrainResolverOption
+    {
+    }
+
+    /// <summary>
+    /// <para>
+    /// Down the part that does not contain the oldest member (current singleton).
+    /// </para>
+    /// When <see cref="DownIfAlone"/> is <c>true</c>:
+    /// <list type="bullet">
+    ///   <item>If the oldest node crashes the others will remove it from the cluster.</item>
+    ///   <item>If oldest node is partitioned from all other nodes, the oldest will down itself and keep all other nodes running.</item>
+    ///   <item>The strategy will not down the single oldest node when it is the only remaining node in the cluster.</item>
+    /// </list>
+    /// When <see cref="DownIfAlone"/> is <c>false</c> and the oldest node crashes, all other nodes will down themselves,
+    /// i.e. shutdown the whole cluster together with the oldest node.
+    /// </summary>
+    public sealed class KeepOldestOption : SplitBrainResolverOption
+    {
+        /// <summary>
+        /// Enable downing of the oldest node when it is partitioned from all other nodes
+        /// </summary>
+        public bool DownIfAlone { get; set; } = true;
+    }
+
+    /// <summary>
+    /// Keep the part that can acquire the lease, and down the other part.
+    /// Best effort is to keep the side that has most nodes, i.e. the majority side.
+    /// This is achieved by adding a delay before trying to acquire the lease on the
+    /// minority side.
+    /// </summary>
+    public sealed class LeaseMajorityOption : SplitBrainResolverOption
+    {
+        /// <summary>
+        /// A class type that extends the abstract class <see cref="Lease"/>
+        /// </summary>
+        public Type LeaseImplementation { get; set; }
```

Comment:
Do you want to refer to the concrete type itself here and extract the FQN later? That way we can statically type this.

Reply:
I don't think we can do that since this type is being instantiated inside `Akka.Coordination` and there's no good way for user to instantiate it during setup. I could add a helper function `WithLeaseImplementation<T>() where T: Lease` to help typing.

Reply:
Yeah a generic method like that may not be a bad idea - that's why I think it's worth playing with this SBR API some more to see how it feels.

Reply:
Scrapped this, I was wrong, `LeaseMajority.LeaseImplementation` is an absolute HOCON path to the lease configuration block, not the implementation class type.